rcl/test/rcl/test_events.cpp

```diff
@@ -730,6 +730,46 @@ TEST_F(TestEventFixture, test_bad_get_handle)
   EXPECT_EQ(NULL, rcl_event_get_rmw_handle(NULL));
 }
 
+/*
+ * Basic test subscriber event message lost
+ */
+TEST_F(TestEventFixture, test_sub_message_lost_event)
+{
+  const rmw_qos_profile_t subscription_qos_profile = default_qos_profile;
+
+  rcl_ret_t ret = setup_subscriber(subscription_qos_profile);
+  ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+  // Check if supported
+  subscription_event = rcl_get_zero_initialized_event();
+  ret = rcl_subscription_event_init(
+    &subscription_event,
+    &subscription,
+    RCL_SUBSCRIPTION_MESSAGE_LOST);
+  EXPECT_TRUE(ret == RCL_RET_OK || ret == RCL_RET_UNSUPPORTED);
+
+  if (ret == RCL_RET_UNSUPPORTED) {
+    // clean up and exit test early
+    ret = rcl_event_fini(&subscription_event);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_subscription_fini(&subscription, this->node_ptr);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
```

Comment:
If you use `OSRF_TESTING_TOOLS_SCOPE_EXIT(...)`, you can avoid the duplicated finalization

Reply:
Is not a duplicated finalization, it's a fini for the event and one for the subscription. The code will be the same inside a `SCOPE_EXIT` wrapper. Am I missing something from your comment?

Reply:
You're also finishing the `event` and `subscription at the end of the test.
If you use the scope exit magic, you will avoid the repetition.

Reply:
Good catch, thanks for the advice!

Reply:
Added with ff94914